jwst/mrs_imatch/mrs_imatch_step.py

```diff
@@ -328,6 +329,20 @@ def _match_models(models, channel, degree, center=None, center_cs='image'):
     #if np.isnan(sigma_data).any():
     #    print('a nan exists in sigma data')
 
+    # MRS fields of view are small compared to source sizes,
+    # and undersampling produces significant differences
+    # in overlap regions between exposures.
+    # Therefore use sigma-clipping to detect and remove sources
+    # (and unmasked outliers) prior to doing the background matching.
+    # Loop over input exposures
+    for image, mask in zip(image_data, mask_data):
+        # Do statistics wavelength by wavelength
+        for thisimg, thismask in zip(image, mask):
+            # Sigma clipped statistics, ignoring zeros where no data
+            _, themed, clipsig = sigclip(thisimg, mask_value=0.)
+            # Reject beyond 3 sigma
+            reject = np.where(np.abs(thisimg - themed) > 3 * clipsig)
```

Comment:
Is `where` really needed?
```suggestion
            reject =np.abs(thisimg - themed) > 3 * clipsig
```

Reply:
It is not _required_ for the logic to function but it may help with performance.For example, if you just leave `reject` as a full boolean array, then when you do something like `data[reject] = 0` it will have to go through all elements of the `data` array. On the other hand, if only a small number of elements are "rejected", then the length of the indices produced by `np.where` will be small and so `data[reject]` will be much faster because it would access only a small number of elements in the `data` array.

Reply:
So, _it depends_ on whether you expect a large number of elements to be rejected (then do not use `np.where`) or a small number (do use `np.where`).

Reply:
Yes, that makes sense. But `np.where` also has its own overhead but maybe insignificant for your use case. Probably premature optimization anyway, so it is fine to leave as is. Thanks for the clarification!